Shared/Samples/Create and edit geometries/CreateAndEditGeometriesView.swift

```diff
@@ -21,15 +21,24 @@ struct CreateAndEditGeometriesView: View {
     @State private var map = Map(basemapStyle: .arcGISTopographic)
     
     /// The model that is required by the menu.
-    @StateObject var model = GeometryEditorMenuModel(
-        geometryEditor: GeometryEditor(),
-        graphicsOverlay: GraphicsOverlay(renderingMode: .dynamic)
-    )
+    @StateObject var model = GeometryEditorMenuModel()
     
     var body: some View {
         VStack {
             MapView(map: map, graphicsOverlays: [model.graphicsOverlay])
                 .geometryEditor(model.geometryEditor)
+                .task {
+                    for await geometry in model.geometryEditor.$geometry {
+                        // Update geometry when there is an update.
+                        model.onGeometryChanged(geometry)
```

Comment:
For both this and `onSelectedElementChanged`, the names are a bit odd. Obviously if a new value is being set, it is because the value changed. Since the change happened in the view, not the model, a more appropriate name would just be `setGeometry(_:)` or `setSelection(_:)`. However, since the implementation of both of those methods simply sets a property on the model, it would make sense just to make those properties settable by the view and do away with the methods that aren't really pulling their own weight.

With all that being said, as far as I can tell, the model doesn't actually even use the selection, which is a pretty good indication that the property should be on a view instead of on the model. And since it is only used by `GeometryEditorMenu`, it seems like the selection should be on that view:

```swift
@State private var selectedElement: GeometryEditorElement?
```

And the associated task modifier should be as well.

It is likely the same should be done for `geometry`; I just haven't dug into it enough to know for sure. While one might argue that the model is "using" it due to it having a property observer, whatever is being done in that property observer could also be done in the associated view. In my opinion, that's not a strong enough argument to put the property in the model. Remember, observable properties on a model equal less efficient view updates. The view model should have only what is necessary.

Reply:
This is good suggestion. I'll apply and refactor the sample a bit.